src/main/java/ar/com/synergian/wagongit/Utils.java

```diff
@@ -11,15 +11,14 @@
 	private Utils() {
 	}
 
-	public static File createCheckoutDirectory(String path, boolean isPermanent) throws GitException {
+	public static File createCheckoutDirectory(String path) throws GitException {
 		File dir;
-		if (isPermanent) {
-			dir = new File(path);
-		} else {
+		if (path.endsWith(".git")) {
 			dir = new File(System.getProperty("java.io.tmpdir"), "wagon-git-" + hashPath(path));
+		} else {
+			dir = new File(path);
```

Comment:
I'm not sure of what you mean =/
Remote is not passed to Utils (it is defined inside GitBackend which calls Utils). So without passing more arguments to the function, apart from the current proposition, i don't see how i can compute the temp file (from remote) and override it with the new argument.
Do you mean something like this :
```
	public static File createCheckoutDirectory(String remote, String overridePath) throws GitException {
		File dir;
		if (overridePath != null && !"".equals(overridePath)) {
			dir = new File(overridePath);
		} else {
			dir = new File(System.getProperty("java.io.tmpdir"), "wagon-git-" + hashPath(remote));
		}
		dir.mkdirs();
		return dir;
	}
```